File: frontend/src/app/core/services/advice.service.ts
import { Injectable, inject } from '@angular/core';
import { Observable } from 'rxjs';

import { ApiService } from './api.service';
import { AdviceResponseModel } from '../models/advice.model';
import { ChatSessionSummary, ChatSessionMessage, PatientMemoryFact } from '../models/chat-session.model';

@Injectable({ providedIn: 'root' })
export class AdviceClientService {
  private readonly api = inject(ApiService);

  ask(
    question: string,
    metricNames?: string[],
    days?: number,
    language?: 'es' | 'en',
    sessionId?: number,
  ): Observable<AdviceResponseModel> {
    return this.api.post<AdviceResponseModel>('/advice', {
      question,
      metricNames,
      days,
      language,
      session_id: sessionId,
    });
  }

  getSessions(): Observable<ChatSessionSummary[]> {
    return this.api.get<ChatSessionSummary[]>('/chat/sessions');
  }

  createSession(): Observable<ChatSessionSummary> {
    return this.api.post<ChatSessionSummary>('/chat/sessions', {});
  }

  getMessages(sessionId: number): Observable<ChatSessionMessage[]> {
    return this.api.get<ChatSessionMessage[]>(`/chat/sessions/${sessionId}/messages`);
  }

  deleteSession(sessionId: number): Observable<unknown> {
    return this.api.delete<unknown>(`/chat/sessions/${sessionId}`);
  }

  getMemory(): Observable<PatientMemoryFact[]> {
    return this.api.get<PatientMemoryFact[]>('/chat/memory');
  }

  deleteMemory(factId: number): Observable<unknown> {
    return this.api.delete<unknown>(`/chat/memory/${factId}`);
  }
}

File: mobile/lib/services/advice_service.dart
import '../core/api_client.dart';
import '../models/advice.dart';

class AdviceService {
  static final _dio = ApiClient.instance;

  static Future<AdviceResponse> ask(
    String question, {
    List<String>? metricNames,
    int days = 180,
    String language = 'es',
  }) async {
    final body = <String, dynamic>{
      'question': question,
      'days': days,
      'language': language,
    };
    if (metricNames != null && metricNames.isNotEmpty) {
      body['metricNames'] = metricNames;
    }
    final res = await _dio.post('/advice', data: body);
    return AdviceResponse.fromJson(res.data as Map<String, dynamic>);
  }
}
